Intersect.Server/Entities/BankInterface.cs

```diff
@@ -194,55 +194,26 @@ public List<int> FindItemSlots(Guid itemId, int quantity = 1)
         /// </summary>
         /// <param name="item">The <see cref="Item"/> to check for.</param>
         /// <returns>Returns whether or not the bank instance can store this item.</returns>
-        public bool CanStoreItem(Item item)
+        private bool CanStoreItem(Item item)
         {
-            if (item.Descriptor != null)
+            // If this item isn't stackable we just need to find an open slot.
+            if (!item.Descriptor.Stackable)
             {
-                // Is the item stackable?
-                if (item.Descriptor.IsStackable)
-                {
-                    // Does the bank have this item already?
-                    var slot = FindItemSlot(item.ItemId);
-                    if (slot >= 0)
-                    {
-                        var existingSlot = mBank[slot];
-
-                        // Can we blindly add more to this stack?
-                        var untilFull = item.Descriptor.MaxBankStack - existingSlot.Quantity;
-                        if (untilFull >= item.Quantity)
-                        {
-                            return true;
-                        }
+                return FindOpenSlot() != -1;
+            }
 
-                        // Check to see if we have the inventory spaces required to hand these items out AFTER filling up the existing slot!
-                        var toGive = item.Quantity - untilFull;
-                        if (Math.Ceiling((double)toGive / item.Descriptor.MaxBankStack) <= FindOpenSlots().Count)
-                        {
-                            return true;
-                        }
-                    }
-                    // User doesn't have this item yet.
-                    else
-                    {
-                        // Does the user have enough free space for these stacks?
-                        if (Math.Ceiling((double)item.Quantity / item.Descriptor.MaxBankStack) <= FindOpenSlots().Count)
-                        {
-                            return true;
-                        }
-                    }
-                }
-                // Not a stacking item, so can we contain the amount we want to give them?
-                else
+            // If it is stackable we need to find an open slot or an existing slot that contains this item type.
+            for (var i = 0; i < mMaxSlots; i++)
+            {
+                var bankSlot = mBank[i];
+                if (bankSlot.ItemId == item.ItemId &&
+                    bankSlot.Quantity + item.Quantity <= bankSlot.Descriptor.MaxBankStack)
                 {
-                    if (FindOpenSlots().Count >= item.Quantity)
-                    {
-                        return true;
-                    }
+                    return true;
                 }
```

Comment:
You removed partial stack filling? If you are trying to put in 100 stackable items, the limit per stack is 100, and you have no totally empty slots but you have two slots that only have 50 which it would fit in, your code rejects it.

Reply:
you are right, this happens if there's no slot left that's empty but for example, two slots that aren't completly filled, i've just added the following after the for loop and it seems to handle things well now:

![imagen](https://user-images.githubusercontent.com/17498701/233818598-0958da9a-8043-41d4-882c-2e70dd0172bd.png)

preview:
https://user-images.githubusercontent.com/17498701/233818612-3a07865c-5033-4316-a31e-2e1f7b47d7dc.mp4

Reply:
Also: with this, we don't really need the for loop that's before this one condition anymore.